### PR TITLE
docs: Feature design template update

### PR DESCRIPTION
**Pull request by solid-vladlaba**
Explain the Analysis section with subsections to simplify feature analysis in the future.

## Checklist

- [x] Documented

### CODE REVIEW

**Comment by solid-vladlaba** [comment]
I have read the CLA Document and I hereby sign the CLA

**Comment by solid-vladlaba** [comment]
recheck

File: docs/19_feature_design_template.md [hunk 1]
```diff
@@ -1,45 +1,79 @@
...
+_**Note**: Prototypes must cover the possible problems and edge cases for the feature, and exemplify how to solve them. There is no need to deep dive into such cases - the [Design](#design) section should discover details. Instead, prototypes should expose that the selected approach provides a convenient way to handle edge cases._
+
+### System modeling
 > Create an abstract model of the system/feature.
 
+The purpose of this section is to discover the relation of the new feature to the existing project's implementations. In general, the section should examine the feature's place in the project's world. To archive this goal, one should consider the existing components of the project and think of the feature as such a component. Then, answer the following:
+
+- Is the new feature a part of the existing component? Does it complement or improve/upgrade this component?
+- Is the new feature should be a part of a standalone component? How this component is going to relate to the project and other components?
+- In what way the new feature integrates into the project's functionality?
```
Comment (illia-romanenko) [inline docs/19_feature_design_template.md:63]:
```suggestion
- Is the new feature a part of the existing component? Does it complement or improve/upgrade this component?
- Is the new feature should be a part of a standalone component? How this component is going to interact with other components?
```

will this be sufficient?

**Review by illia-romanenko — CHANGES_REQUESTED** [review]
Great job capturing it all! I've made a couple of suggestions but overall it could benefit significantly in other team members understanding this template, thanks @solid-vladlaba !

File: docs/19_feature_design_template.md [hunk 2]
```diff
@@ -1,45 +1,96 @@
...
+In this section, describe the methodology to feature analysis, possible solutions, and areas that need to be explored, ways to compare alternatives. It should be a brief plan for all sub-sections.
 
-### Feasibility study
+_**Important Note**: Avoid excessive details in the Analysis subsections. Think about the feature at the very top-level abstraction. Provide the details only where necessary (e.g. in the [Prototyping](#prototyping) section)._
 
+### Feasibility study
 > A preliminary study of the feasibility of implementing this feature.
 
-### Requirements
+Once the requirements are defined, one should consider the feature and conclude if this feature makes sense for the project. In the context of usability and project's purposes, one should answer the following:
```
Comment (illia-romanenko) [inline docs/19_feature_design_template.md:30]:
```suggestion
One should answer the following: 
```

File: docs/19_feature_design_template.md [hunk 3]
```diff
@@ -1,45 +1,96 @@
...
 > A preliminary study of the feasibility of implementing this feature.
 
-### Requirements
+Once the requirements are defined, one should consider the feature and conclude if this feature makes sense for the project. In the context of usability and project's purposes, one should answer the following: 
+
+- How this feature would help end-users and would they use it?
+- Does this feature meet the project's scope?
+- Does this feature fit the project's purpose?
 
+Once the feature makes sense, one should review is it feasible. More precisely, one should conclude is it possible to implement the feature taking into account the listed requirements from the [Requirements](#requirements) section. It is possible here to refer to existing solutions in the [Landscape](#landscape) section.
```
Comment (illia-romanenko) [inline docs/19_feature_design_template.md:36]:
```suggestion
```

Let's skip references to the future sections, I think for now we can limit to the above questions.

**Review by illia-romanenko — CHANGES_REQUESTED** [review]
Looks great, thanks @solid-vladlaba ! Please see if couple additional items make sense and that should be it. Thanks for leading this effort.

**Review by illia-romanenko — APPROVED** [review]
LGTM - thanks @solid-vladlaba !

**Pull request merged**